### DISCUSSION TITLE
How to change the RUN_TIME version of deployed store procedures?

### DISCUSSION THREAD

**Discussion by nicosan-eos** [Snowflake-Labs/snowcli#1647] [Q&A]
I have been using Snowflake CLI to orchestrate and deploy store procedures mostly.

I have my app setup properly in my local repo
![image](https://github.com/user-attachments/assets/de512812-c249-4cf9-a39a-764739702e0a)

Everything works as expected. To deploy my app, I just use the `snow snowpark deploy --replace` command.

This is an example of one of the store procedure I have deployed (which works perfectly):

![image](https://github.com/user-attachments/assets/0e327fed-e0fe-4977-b464-2bbd28d33a6b)

However, I have come across the following information about Python runtime_version support:

> - Stage One: Starting October 14, 2024, Snowflake will no longer apply security patches or other updates to the Python 3.8 runtime, and objects using the Python 3.8 runtime will no longer be eligible for technical support. 
> - Stage Two: Starting March 31, 2025, you will no longer be able to create new objects using Python 3.8. 

I tried to search in the documentation available a way of changing the default python's runtime version for the deployed store procedures, but I was not able to find anything. I tried to execute `snow snowpark deploy --env RUNTIME_VERSION='3.9' --replace` command instead, but it didn't work.

Is there a way we can change the runtime version?

See: 
[1] [Status of Python versions](https://go.snowflake.net/MjUyLVJGTy0yMjcAAAGVpL1-R0FgmgNmlFJunVt0j-R48F8InXsbYEH5vCWrtWbMOo-03xQVZhJl4DS8V1K3T46I-q4=) 
[2] [Snowflake Python Runtime Support](https://go.snowflake.net/MjUyLVJGTy0yMjcAAAGVpL1-SDQkGDLyBiaUDWjRQfnTv1T0SdPX3GbqWbl429hvneo1zngoEdgUr2KCt22MVswV2ks=)

**Comment by sfc-gh-turbaszek** [comment]
@nicosan-eos you should upgrade the runtime version in your `snowflake.yml` file
```
runtime_version: "3.9"
```
